docker-compose.yml

```
@@ -0,0 +1,30 @@
+version: "3"
+
+services:
+  bot:
+    build: .
+    restart: always
+    privileged: true
+    links:
+      - redis
+    networks:
+      - internal_network
+      - external_network
+    environment:
+      - PETROL_TOKEN
+
+  redis:
+    restart: always
+    image: redis:6.0.4-alpine
```

```suggestion
    image: redis:6.0.4-alpine
    deploy:
        restart_policy:
            condition: any
            max_attempts: 3
```
You haven't set `max_attempts`, which means docker-compose never gives up.
You can set some value on it so that it gives up after some retries.